filters/tracing/spanname.go

```diff
@@ -0,0 +1,53 @@
+/*
+Package tracing provides filters to instrument distributed tracing.
+*/
+package tracing
+
+import (
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	// SpanNameFilterName is the name of the filter in eskip.
+	SpanNameFilterName = "tracingSpanName"
+
+	// OpenTracingProxySpanKey is the key used in the state bag to pass the span name to the proxy.
+	OpenTracingProxySpanKey = "statebag:opentracing:proxy:span"
+)
+
+type spec struct{}
+
+type filter struct {
+	spanName string
+}
+
+// NewSpanName creates a filter spec for setting the name of the outgoing span. (By default "proxy".)
+//
+// 	tracingSpanName("example-operation")
+//
+// WARNING: this filter is experimental, and the name and the arguments can change until marked as stable.
```

Comment:
Do we have a policy documented somewhere how to mark things as experimental or deprecated?
I like the idea, maybe it's enough like this and we should create a separate issue to document how to do this with introducing new features and how to deal with old stuff, which I personally mark as DEPRECATED in our code base.

Reply:
we don't have currently such a policy, and yes it would be useful. I like the godoc for this, because that's something that we can assume someone always reads. How about using all-caps labels: EXPERIMENTAL, DEPRECATED, and the reason, in its own paragraph of the related godoc? e.g:

```
// NewFoo creates and initializes a new foo object. Use the Close() method to release
// resources, once it's not required anymore.
//
// EXPERIMENTAL: this feature is in an experimental state, and can be removed or
// changed in the near future.
//
func NewFoo() foo { return 42 }
```

Reply:
I like EXPERIMENTAL (better than WARNING) and DEPRECATED as godoc